### ISSUE TITLE
how to reproduce memory snapshot in doc？

### ISSUE THREAD

**Issue by LucQueen** [pytorch-labs/segment-anything-fast#112]
hi，how to reproduce memory snapshot in doc？
![image](https://github.com/pytorch-labs/segment-anything-fast/assets/32257212/335b2de1-cd3c-492d-b197-15441b593bd9)

what i get is
![image](https://github.com/pytorch-labs/segment-anything-fast/assets/32257212/514710bc-9ba8-4f0d-a92a-1507aa726e8b)

I‘m very confused why can not get ‘add_decomposed_rel_pos’ stack informations in memory snapshot. The torch version is 2.2

**Comment by cpuhrsch** [comment]
Hi @LucQueen - which GPU type are you running on? Thank you.

**Comment by LucQueen** [comment]
@cpuhrsch thanks for your reply. The GPU type is A100 80G SXM.

**Comment by cpuhrsch** [comment]
@LucQueen - oh ok! That's probably because you're using the fused kernel. So add_decomposed_rel_pos has been shortened and now doesn't materialize the full attention mask instead. Instead we're using `flash_4` to fuse the construction, which is a lot more memory efficient. The reason you're seeing a bigger memory footprint is probably because you're using a larger batch size than from that snapshot. That snapshot is from the unmodified segment-anything.

See https://github.com/pytorch-labs/segment-anything-fast/blob/387488bc4c7ab2ae311fb0632b34cab5cbfbab78/segment_anything_fast/modeling/image_encoder.py#L233-L247

**Comment by LucQueen** [comment]
@cpuhrsch thanks for your reply! I am using batch-size 16, whereas doc is using batch-size 8, so I am seeing a bigger memory footprint. But I'm still very confused, I see memory snapshot in doc is also using fused kernel, you can see it by marked in red box from the picture, why I can not get ‘add_decomposed_el_pos’ stack informations in memory snapshot.
![image](https://github.com/pytorch-labs/segment-anything-fast/assets/32257212/bdf9f705-4372-44c5-9c69-0436abbd4541)

**Comment by cpuhrsch** [comment]
@LucQueen - Ah! Hm, I'm not sure. Is your picture from the latest version of segment-anything-fast?

The picture you reference is [from a section within the blog](https://pytorch.org/blog/accelerating-generative-ai/#triton-custom-sdpa-for-fused-relative-positional-encoding) and not based on the most recent version of segment-anything-fast. It was recorded from an earlier version without the fused kernels.
